htdocs/accountancy/class/bookkeeping.class.php

```diff
@@ -760,7 +789,7 @@ public function fetch($id, $ref = null, $mode = '')
 		$sql .= ' WHERE 1 = 1';
 		$sql .= " AND entity = " . ((int) $conf->entity); // Do not use getEntity for accounting features
 		if (null !== $ref) {
-			$sql .= " AND t.rowid = ".((int) $ref);
+			$sql .= " AND t.ref = '".$this->db->escape($ref)."'";
```

Comment:
Currently, we have no field ref (we don't really need it), so we use the id as unique ref key. So old test looks as the good version

Reply:
Ok, old method restored